modules/c++/nitf/tests/strip_invisible_segments.cpp

```diff
@@ -0,0 +1,159 @@
+/* =========================================================================
+ * This file is part of NITRO
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2017, MDA Information Systems LLC
+ *
+ * NITRO is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <mem/SharedPtr.h>
+#include <import/nitf.hpp>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <memory>
+
+// Round-trip a NITF, removing any image segments with an IREP of NODISPLY
+namespace
+{
+void stripImages(nitf::Record& record)
+{
+    std::vector<size_t> invisibleImages;
+    nitf::List images = record.getImages();
+    nitf::ListIterator iter = images.begin();
+    for (size_t ii = 0; iter != images.end(); ++ii, ++iter)
+    {
+        nitf::ImageSegment image = *iter;
+        nitf::ImageSubheader subheader = image.getSubheader();
+        std::string irep(subheader.getImageRepresentation().getRawData());
```

Comment:
You don't need the `.getRawData()` here do you?

Reply:
Turns out you do need it for the copy ctor, but not for the assignment operator.